pimd/pim_msdp_packet.c

```diff
@@ -441,7 +444,7 @@ void pim_msdp_pkt_sa_tx(struct pim_instance *pim)
 
 void pim_msdp_pkt_sa_tx_one(struct pim_msdp_sa *sa)
 {
-	pim_msdp_pkt_sa_fill_hdr(sa->pim, 1 /* cnt */);
+	pim_msdp_pkt_sa_fill_hdr(sa->pim, 1 /* cnt */, sa->rp);
```

Comment:
You are right. If the router is initiating a SA message then it would be both the originator and and RP in that message. Whereas, if the router is forwarding a SA message, then it will be forwarding a messge originated in another RP. If we assume that this function will aways be used to send SA messages originated on this router, then it is safe to replace "sa->rp" with "pim->msdp_originator_id".